jsapp/js/components/common/tabs.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {useState} from 'react';
+import styles from './tabs.module.scss';
+
+interface Tab {
+  label: string;
+  route: string;
+}
+
+export interface TabsProps {
+  tabs: Tab[];
+  selectedTab: string;
+  onChange: (route: string) => void;
+}
+
+export default function Tabs({tabs, selectedTab, onChange}: TabsProps) {
+  const [activeTab, setActiveTab] = useState(selectedTab);
+
+  const handleTabKeyDown = (event: React.KeyboardEvent<HTMLButtonElement>) => {
+    if (event.key === 'Tab') {
+      event.preventDefault();
+
+      const currentIndex = tabs.findIndex((tab) => tab.route === activeTab);
+
+      const nextIndex = (currentIndex + 1) % tabs.length;
+
+      setActiveTab(tabs[nextIndex].route);
+      onChange(tabs[nextIndex].route);
+    }
+  };
+
+  const renderTab = (tab: Tab) => {
+    return {
+      key: tab.route,
+      onClick: () => {
+        setActiveTab(tab.route);
+        onChange(tab.route);
+      },
+      onKeyDown: (event: React.KeyboardEvent<HTMLButtonElement>) =>
+        handleTabKeyDown(event),
+      className: `${styles.tab} ${
+        activeTab === tab.route ? styles.active : ''
+      }`,
+      'aria-selected': activeTab === tab.route,
+      tabIndex: activeTab === tab.route ? 0 : 1,
```

Comment:
The [tabIndex for inactive tabs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#best_practices) should be `-1` - currently, you can't tab out of the tab component to focus on the rest of the page.